### PR TITLE
feat: add serverless api for hemi sepolia

### PR DESCRIPTION
**Pull request by gndelia**
This PR creates a serverless api to call from uniswap's frontend to get a quote price. A few simplifications were made:

- subgraph providers were removed
- token list is hardcoded instead of using a json file in S3
- Simulations removed 

As the original code does not use serverless and uses CDK with custom secrets that are not available to us, I found that creating a wrapper API with serverless to be the easiest way to get away with getting this REST API working.

The frontend sends a different structure (because it actually calls [the unified routing API](https://github.com/uniswap/unified-routing-api)) which then forwards into this API, so the serverless API does the transformations needed. Eventually, we may want to also add the `unified-routing-api` , which will also require using `serverless` 😄

### CODE REVIEW

**Review by leanazulyoro — DISMISSED** [review]

**Review by jcvernaleo — DISMISSED** [review]
OK

File: lib/handlers/router-entities/hardcoded-token-list-provider.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    chainId: hemi.id,
+    decimals: hemi.nativeCurrency.decimals,
+    name: hemi.nativeCurrency.name,
+    symbol: hemi.nativeCurrency.symbol,
+  },
+]
+
+const tokenList: TokenList = {
+  keywords: ['default', 'uniswap'],
+  name: 'Uniswap Labs Default',
```
Comment (gabmontes) [inline lib/handlers/router-entities/hardcoded-token-list-provider.ts]: Should we update these two?

File: api/serverless.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      pattern: ['./api/**/*.ts', '../lib/**/*.ts']
+  serverless-offline:
+    httpPort: 4000
+
+functions:
+  get-quote:
+    events:
+      - http: 
+          method: post
+          path: /v2/quote
```
Comment (gabmontes) [inline api/serverless.yml:18]: Don't we need CORS?

File: api/index.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+})
+
+let quoteHandler: QuoteHandler
+try {
+  const quoteInjectorPromise = new QuoteHandlerInjector('quoteInjector').build()
+  quoteHandler = new QuoteHandler('quote', quoteInjectorPromise)
+} catch (error) {
+  log.fatal({ error }, 'Fatal error')
+  throw error
+}
```
Comment (gabmontes) [inline api/index.ts]: Is this `try/catch` needed?

File: api/serverless.yml (hunk 2 above)
Comment (leanazulyoro) [reply api/serverless.yml:18]: It's actually setting `Access-Control-Allow-Origin` as `*` in the original handler. I guess once we know the url the interface will be deployed on we might set it to that.

File: lib/handlers/router-entities/hardcoded-token-list-provider.ts (hunk 1 above)
Comment (leanazulyoro) [reply lib/handlers/router-entities/hardcoded-token-list-provider.ts]: This doesn't change anything, but we can change to "hemi". Here: https://github.com/hemilabs/routing-api/pull/1/commits/9f64f9e472217c436c8397d886494f418b7b6009

File: api/serverless.yml (hunk 2 above)
Comment (leanazulyoro) [reply api/serverless.yml:18]: Additionally, I added `cors: true` on serverless config.

File: api/index.ts (hunk 3 above)
Comment (leanazulyoro) [reply api/index.ts]: I moved it inside the handler, and return a 500 error if anything fails. here: https://github.com/hemilabs/routing-api/pull/1/commits/2115f36ced28fba7ff88616347d7ed5bff3d8c3b

**Review by gabmontes — DISMISSED** [review]

**Review by gabmontes — APPROVED** [review]

**Pull request merged**